uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/app.js

```diff
@@ -29,10 +35,16 @@ define([
     'common.models',
     'my-leave.config',
     'my-leave.components',
+    'leave-absences.directives',
     'leave-absences.models',
+    'common.angularDate',
   ])
-  .run(['$log', function ($log) {
+  .run(['$log','DateFormat', function ($log, DateFormat) {
     $log.debug('app.run');
+    //set HR_Settings
+    DateFormat.getDateFormat().then(function(result){
```

Comment:
Two things here:

1. The `then` function is useless and confusing, since you're not doing anything with it
2. `HR_Settings.DATE_FORMAT` is already assigned in the [`my-leave.config`](https://github.com/civicrm/civihr/blob/f9cb7301357a0e47bd363663c05fa6ef1dad83a9/uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/my-leave/modules/config.js#L27) module, an the reason for doing it in the `resolve` property is that until the value is set, the application is not loaded. Instead what you are doing here is calling an async operation in the `.run()` method, basically creating a race condition, hoping that it fetched the `DATE_FORMAT` value before it is actually used anywhere in your code